app/frontend/wallet/shelley/shelley-ledger-crypto-provider.ts

```diff
@@ -64,7 +64,9 @@ import {TxRelayType, TxStakepoolOwner, TxStakepoolRelay} from './helpers/poolCer
 import assertUnreachable from '../../helpers/assertUnreachable'
 
 let _activeTransport: Transport | null = null
-const getLedgerTransport = async (ledgerTransportType: LedgerTransportType): Promise<Transport> => {
+const getLedgerTransport = async (
+  ledgerTransportType: LedgerTransportType
+): Promise<Transport | null> => {
```

Comment:
this code should not deal with any other ledger transport type than the ones handled here (webhid, u2f, webusb) and should throw in such case, so it should either throw or return a proper Transport. The "default" option matters really just for the UI. I would propose refactoring it like this, even within this PR as it shouldn't be much work and would be less confusing than relaxing the type signature of this function

1. Introduce a `LedgerTransportChoice` type which would be equivalent to the current `LedgerTransportType` and rename all existing instances in the code to LedgerTransportChoice
2. introduce `type LedgerTransportType = Exclude<LedgerTransportChoice, LedgerTransportChoice.DEFAULT>` and migrate from `LedgerTransportChoice` to `LedgerTransportType` in `getLedgerTransport()` output type  and `getDefaultLedgerTransportType()` function
3. `getLedgerTransport()` now can be refactored to `return assertUnreachable` in the default branch of the switch statement